File: verification/verify_whats_new.ts
import { _electron as electron } from 'playwright';
import path from 'path';
import fs from 'fs';

async function run() {
  // Launch Electron application
  const electronApp = await electron.launch({
    args: ['.'],
    env: {
       ...process.env,
       NODE_ENV: 'development'
    }
  });

  // Get the first window
  const window = await electronApp.firstWindow();

  // Wait for app to load
  await window.waitForLoadState('domcontentloaded');

  try {
    // 1. Check if "What's New" dialog is visible
    const dialogTitle = window.getByRole('heading', { name: "What's New" });

    if (!await dialogTitle.isVisible()) {
        console.log("What's New dialog not initially visible. Looking for trigger...");
        await window.waitForTimeout(2000);
    }

    if (await dialogTitle.isVisible()) {
        console.log("What's New dialog found!");

        // Take a screenshot of the main grid view
        await window.screenshot({ path: 'verification/whats_new_grid.png' });

        // 2. Click "Recent Highlights"
        await window.getByText('Recent Highlights').click();
        await window.waitForTimeout(500); // Wait for transition
        await window.screenshot({ path: 'verification/recent_highlights.png' });

        // Go back - target the button inside the WhatsNewDialog
        // There is a back button in the header
        await window.locator('button[aria-label="Back"]').first().click();
        await window.waitForTimeout(500);

        // 3. Click "Getting Started"
        await window.getByText('Getting Started').click();
        await window.waitForTimeout(500);
        await window.screenshot({ path: 'verification/getting_started.png' });

        // Go back
        await window.locator('button[aria-label="Back"]').first().click();
        await window.waitForTimeout(500);

        // 4. Click "Full Changelog" button (in the grid)
        await window.getByRole('button', { name: 'Full Changelog' }).click();
        await window.waitForTimeout(500);
        await window.screenshot({ path: 'verification/full_changelog.png' });

        // Switch back to home
         await window.locator('button[aria-label="Back"]').first().click();
         await window.waitForTimeout(500);
         await window.screenshot({ path: 'verification/back_to_home.png' });

    } else {
        console.log("Could not find What's New dialog. Taking screenshot of main window.");
        await window.screenshot({ path: 'verification/main_window.png' });
    }

  } catch (error) {
    console.error("Error during verification:", error);
    await window.screenshot({ path: 'verification/error.png' });
  } finally {
    await electronApp.close();
  }
}

run();

File: verification/verify_whats_new.py
from playwright.sync_api import sync_playwright, expect

def run(playwright):
    browser = playwright.chromium.launch()
    page = browser.new_page()
    page.set_viewport_size({"width": 1280, "height": 720})

    print("Navigating to app...")
    page.goto("http://localhost:4173/")

    # Wait for app to load
    page.wait_for_load_state("networkidle")

    print("Pressing Shift+N to open What's New...")
    page.keyboard.press("Shift+N")

    # Wait for dialog to open
    page.wait_for_selector("#whats-new-title")

    # Check for the new content
    print("Checking for new content...")

    # Try to navigate to "Latest Highlights" if in grid view
    try:
        # Wait a bit for grid to render
        page.wait_for_timeout(1000)

        # Look for the card with "Latest Highlights"
        latest = page.get_by_role("heading", name="Latest Highlights")
        if latest.is_visible():
            latest.click()
            print("Clicked Latest Highlights")
    except Exception as e:
        print(f"Navigation note: {e}")

    # Now check for the content text headers
    # Live Code Preview is an H3 in markdown, so likely rendered as H3
    expect(page.get_by_role("heading", name="Live Code Preview")).to_be_visible()
    expect(page.get_by_role("heading", name="Export Improvements")).to_be_visible()

    # Check for specific text content
    expect(page.get_by_text("Real-time Code Generation")).to_be_visible()
    expect(page.get_by_text("Direct Download")).to_be_visible()

    print("Taking screenshot...")
    page.screenshot(path="verification/whats_new.png")

    browser.close()

if __name__ == "__main__":
    with sync_playwright() as playwright:
        run(playwright)
